hydra-node/src/Hydra/Node/Run.hs

```diff
@@ -80,15 +90,37 @@ run opts = do
       pparams <- case toLedgerPParams ShelleyBasedEraBabbage protocolParams of
         Left err -> throwIO (ConfigurationException err)
         Right bpparams -> pure bpparams
-      withCardanoLedger chainConfig pparams $ \ledger -> do
+      let onlineOrOfflineConfig = case offlineConfig of
+            Nothing -> Right chainConfig
+            Just offlineConfig' -> Left offlineConfig'
+
+      let DirectChainConfig{networkId, nodeSocket} = chainConfig
+
+      globals <- case offlineConfig of
+        Nothing -> do
+          -- online
+          globals' <- newGlobals =<< queryGenesisParameters networkId nodeSocket QueryTip
+          pure globals'
+        Just OfflineConfig{ledgerGenesisFile} -> do
+          -- offline
+          loadGlobalsFromGenesis ledgerGenesisFile
+
+      withCardanoLedger pparams globals $ \ledger -> do
         persistence <- createPersistenceIncremental $ persistenceDir <> "/state"
         (hs, chainStateHistory) <- loadState (contramap Node tracer) persistence initialChainState
+
         checkHeadState (contramap Node tracer) env hs
         nodeState <- createNodeState hs
         -- Chain
-        ctx <- loadChainContext chainConfig party hydraScriptsTxId
-        wallet <- mkTinyWallet (contramap DirectChain tracer) chainConfig
-        withDirectChain (contramap DirectChain tracer) chainConfig ctx wallet chainStateHistory (putEvent . OnChainEvent) $ \chain -> do
+        let withChain cont = case onlineOrOfflineConfig of
+              Left offlineConfig' ->
+                let headId = UnsafeHeadId "HeadId"
+                 in withOfflineChain (contramap DirectChain tracer) offlineConfig' globals headId party contestationPeriod chainStateHistory (putEvent . OnChainEvent) cont
```

Comment:
I would suggest to hard-code the `headId` and `contestationPeriod` in the offline chain. This will have you pass less arguments around.

Also if you follow my other suggestions, `globals` will be unused.